Makefile

```diff
@@ -0,0 +1,6 @@
+init:
+	pip install pipenv
+	pipenv install
+
+test:
```

Comment:
Doesn't test target need the pipenv from the init target?

Reply:
I don't get it. What do mean by `init target` ?

Reply:
I assume you have to execute `pip install pipenv` & `pipenv install` that you can run `pipenv run pytest tests/.`, I'm correct?

Reply:
but I am doing that right ?
I mean the first two lines are doing the installation and then in `test` section I am running the tests

Reply:
If you run `make test`, the commands from `init` target won't get executed.

Reply:
but I am not doing it I mean in .travis.yml I am doing just `install: -make` to run the whole Makefile

Reply:
- you don't execute both targets by calling `make`
- I expect that running test also sets up the environment before executing the tests

Reply:
oh yes just noticed that so I would have to change the `install: make` to

Reply:
oh yes just noticed it so now I'll have to change `install: make` to
```
install:
    - make init
    - make test
```

Reply:
two questions with this change now
- Why do you have to call init before test (if you want to do both in the same step)? Because your Makefile is wrong.
- Why run test in install step?

Reply:
* How is my makefile wrong ?

Reply:
test doesn't have init as dependency.

Reply:
okay now I got. so it should look like:

Reply:
okay so now I got it. It should look like:

`Makefile`
```
init:
    pip install pipenv
    pipenv install 

test:
    pipenv run pytest tests/.
```

`.travis.yml`
```
install:
    - make

sript:
    - make test
```

right ?

Reply:
- travis looks correct, beside that you should keep the additional repo tests
- test target still doesn't depend on init target (you haven't changed anything)

Reply:
* yes those additional repo test will be there
* by test target do you mean instead of `tests/.` I should use just `tests` like  `pipenv run pytest tests`

Reply:
I like having a makefile. @Razzeee it seems you also don't get it.
```
init:
	pip install pipenv
	pipenv install

test: init
	pipenv run pytest tests/.
```

Reply:
All of what I said was from https://docs.pipenv.org/advanced/#travis-ci

Reply:
It's correct for running it with travis, but you may want to run `make test` locally too and don't want to bother running `make init` explicitly first.

Reply:
But even without you're method when I do `make test` they seems to be running. fine.

Reply:
Yes, for local it's fine. But having it like that will make travis do it twice, if we don't omit the init in that file.

Which also leads to me asking myself if there is a cache/performance gain to doing it in the install step.

Reply:
Then it has to be further optimized.